includes/class-wc-admin-api-init.php

```diff
@@ -312,39 +373,52 @@ public static function orders_data_store_init() {
 	}
 
 	/**
-	 * Init orders product lookup store.
-	 *
-	 * @param WC_Background_Updater|null $updater Updater instance.
-	 * @return bool
+	 * Init order/product lookup tables update (in batches).
 	 */
-	public static function order_product_lookup_store_init( $updater = null ) {
-		// TODO: this needs to be updated a bit, as it no longer runs as a part of WC_Install, there is no bg updater.
-		global $wpdb;
+	public static function orders_lookup_batch_init() {
+		$batch_size  = self::get_batch_size( self::ORDERS_BATCH_ACTION );
+		$order_query = new WC_Order_Query(
+			array(
+				'return'   => 'ids',
+				'limit'    => 1,
+				'paginate' => true,
+			)
+		);
+		$result      = $order_query->get_orders();
 
-		$orders = get_transient( 'wc_update_350_all_orders' );
-		if ( false === $orders ) {
-			$orders = wc_get_orders(
-				array(
-					'limit'  => -1,
-					'return' => 'ids',
-				)
-			);
-			set_transient( 'wc_update_350_all_orders', $orders, DAY_IN_SECONDS );
+		if ( 0 === $result->total ) {
+			return;
 		}
 
-		// Process orders until close to running out of memory timeouts on large sites then requeue.
-		foreach ( $orders as $order_id ) {
+		$num_batches = ceil( $result->total / $batch_size );
+
+		self::queue_batches( 1, $num_batches, self::ORDERS_BATCH_ACTION );
+	}
+
+	/**
+	 * Process a batch of orders to update (stats and products).
+	 *
+	 * @param int $batch_number Batch number to process (essentially a query page number).
+	 * @return void
+	 */
+	public static function orders_lookup_process_batch( $batch_number ) {
+		$batch_size  = self::get_batch_size( self::ORDERS_BATCH_ACTION );
+		$order_query = new WC_Order_Query(
+			array(
+				'return'  => 'ids',
+				'limit'   => $batch_size,
+				'page'    => $batch_number,
+				'orderby' => 'ID',
+				'order'   => 'ASC',
+			)
+		);
+		$order_ids = $order_query->get_orders();
+
+		foreach ( $order_ids as $order_id ) {
+			// TODO: schedule single order update if this fails?
+			WC_Admin_Reports_Orders_Data_Store::sync_order( $order_id );
 			WC_Admin_Reports_Products_Data_Store::sync_order_products( $order_id );
```

Comment:
Should this maybe include other lookups as well? Coupons, taxes, etc.

Reply:
Ah, yes - I only handled what was currently being processed in the "renegerate" logic. I'll add the other lookup tables here as well.

Reply:
Added in f788e00.

Reply:
Looking at this, and seeing that downloads data store has no update function, it might be worth noting somewhere that donwloads are not rebuilt using this mechanism at the moment.